pkg/ingester/ingester.go

```diff
@@ -84,6 +84,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.Float64Var(&cfg.SyncMinUtilization, "ingester.sync-min-utilization", 0, "Minimum utilization of chunk when doing synchronization.")
 	f.IntVar(&cfg.MaxReturnedErrors, "ingester.max-ignored-stream-errors", 10, "Maximum number of ignored stream errors to return. 0 to return all errors.")
 	f.DurationVar(&cfg.MaxChunkAge, "ingester.max-chunk-age", time.Hour, "Maximum chunk age before flushing.")
+	f.DurationVar(&cfg.QueryStoreMaxLookBackPeriod, "ingester.query-store-max-look-back-period", 0, "How far back should an ingester be allowed to query the store for data, for use only with boltdb-shipper index and filesystem object store. -1 for infinite.")
```

Comment:
I think the requirement should be `boltdb`(not `boltdb_shipper`) as index store and filesystem as object store because there is no use of having `boltdb_shipper` without shared object store otherwise you would be copying files somewhere on the same filesystem which is essentially same as not using `boltdb_shipper` at all. I might be wrong here or might be missing your point. What do you think?

Reply:
If my above assumption is true, that the file names are unique, I specified `boltdb_shipper` so that you could copy the files between ingesters to allow scaling down, otherwise what you said is true that there isn't much reason to use `boltdb_shipper` for this rather than `boltdb`.

With the exception of one other reason I could think of, it would be easier to take backups or even move the Loki to another host because the index files can be directly copied with the chunks directory while Loki is still running, I don't think this is true for `boltdb` I don't think you can safely copy an open boltdb file.